iOS build failed!

commit id:
```
3a0854ce2104dcf1549d7c46959154e02dbba347
```

log:
```
➜  glutin git:(master) cargo build --target=aarch64-apple-ios
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling khronos_api v2.2.0                                                 
   Compiling cfg-if v0.1.4
   Compiling bitflags v1.0.3
   Compiling libc v0.2.42
   Compiling lazy_static v1.0.1
   Compiling xml-rs v0.7.0
   Compiling log v0.4.3
   Compiling malloc_buf v0.0.6
   Compiling shared_library v0.1.8
   Compiling objc v0.2.3
   Compiling winit v0.16.2
   Compiling gl_generator v0.9.0
   Compiling glutin v0.17.0 (file:///Users/leizh007/Documents/iOS%E5%AD%A6%E4%B9%A0/Rust/glutin)
error[E0050]: method `resize` has 3 parameters but the declaration in trait `GlContext::resize` has 2
   --> src/api/ios/mod.rs:345:44
    |
345 |     fn resize(&self, _width: u32, _height: u32) {
    |                                            ^^^ expected 2 parameters, found 3
    | 
   ::: src/lib.rs:139:28
    |
139 |     fn resize(&self, size: dpi::PhysicalSize);
    |                            ----------------- trait requires 2 parameters

error: aborting due to previous error

For more information about this error, try `rustc --explain E0050`.
error: Could not compile `glutin`.

To learn more, run the command again with --verbose.

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,63 @@
 language: rust
 
-rust:
-  - nightly
-  - beta
-  - stable
-  - 1.24.1
-
 cache: cargo
+
+matrix:
+  include:
+    # Linux 32bit
+    - env: TARGET=i686-unknown-linux-gnu
+      os: linux
+      rust: nightly
+      addons:
+        apt:
+          # Cross compiler and cross compiled C libraries
+          packages: &i686_packages
+            - gcc-multilib
+    - env: TARGET=i686-unknown-linux-gnu
+      os: linux
+      rust: stable
+      addons:
+        apt:
+          packages: *i686_packages
+    - env: TARGET=i686-unknown-linux-gnu
+      os: linux
+      rust: 1.24.1
+      addons:
+        apt:
+          packages: *i686_packages
+
+    # Linux 64bit
+    - env: TARGET=x86_64-unknown-linux-gnu
+      os: linux
+      rust: nightly
+    - env: TARGET=x86_64-unknown-linux-gnu
+      os: linux
+      rust: stable
+    - env: TARGET=x86_64-unknown-linux-gnu
+      os: linux
+      rust: 1.24.1
+
+    # macOS
+    - env: TARGET=x86_64-apple-darwin
+      os: osx
+      rust: nightly
+    - env: TARGET=x86_64-apple-darwin
+      os: osx
+      rust: stable
+    - env: TARGET=x86_64-apple-darwin
+      os: osx
+      rust: 1.24.1
+
+    # iOS
+    - env: TARGET=x86_64-apple-ios
+      os: osx
+      rust: nightly
+    - env: TARGET=x86_64-apple-ios
+      os: osx
+      rust: stable
+    - env: TARGET=x86_64-apple-ios
+      os: osx
+      rust: 1.24.1
 
 addons:
   apt:
@@ -18,13 +69,14 @@ env:
   global:
     - secure: "kVjHgK+6ivT5tUleNH/m6+69tPf8gkPNSPLsnXrkMJaIeXSnnKKQegP1O0UCThYAUbbwrUGlrKJUngDxkD8jSANhzAULX9t1iXho41InzYBEErTanXBYF8WO1rK1keKjLwiYw4KuvbiBoSeKabzQRwv0kMttxY57+hKVROr5dSg="
 
-script:
-  - cargo build --verbose
-  - cargo test --verbose
+install:
+  - rustup self update
+  - rustup target add $TARGET; true
 
-os:
-  - linux
-  - osx
+script:
+  - cargo build --target $TARGET --verbose
+    # Running iOS apps on OSX requires the simulator so we skip that for now
+  - if [ "$TARGET" != "x86_64-apple-ios" ]; then cargo test --target $TARGET --verbose; fi
 
 after_success:
   - |

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 - **Breaking:** Added `OsError` variant to `ContextError`.
 - Improved glX error reporting.
+- The iOS backend no longer fails to compile... again (added iOS testing on CI to prevent further issues).
 
 # Version 0.17.0 (2018-06-27)
 

--- a/src/api/ios/mod.rs
+++ b/src/api/ios/mod.rs
@@ -70,7 +70,6 @@ use {
     CreationError,
     EventsLoop,
     GlAttributes,
-    GlContext,
     GlRequest,
     PixelFormat,
     PixelFormatRequirements,
@@ -267,17 +266,9 @@ impl Context {
             panic!("framebuffer status: {:?}", status);
         }
     }
-}
 
-impl Drop for Context {
-    fn drop(&mut self) {
-        let _: () = unsafe { msg_send![self.eagl_context, release] };
-    }
-}
-
-impl GlContext for Context {
     #[inline]
-    unsafe fn make_current(&self) -> Result<(), ContextError> {
+    pub unsafe fn make_current(&self) -> Result<(), ContextError> {
         let context_class = Class::get("EAGLContext").expect("Failed to get class `EAGLContext`");
         let res: BOOL = msg_send![context_class, setCurrentContext: self.eagl_context];
         if res == YES {
@@ -290,12 +281,13 @@ impl GlContext for Context {
     }
 
     #[inline]
-    fn is_current(&self) -> bool {
-        false
+    pub fn is_current(&self) -> bool {
+        // TODO: This can likely be implemented using `currentContext`/`getCurrentContext`
+        true
     }
 
     #[inline]
-    fn get_proc_address(&self, proc_name: &str) -> *const () {
+    pub fn get_proc_address(&self, proc_name: &str) -> *const () {
         let proc_name_c = CString::new(proc_name).expect("proc name contained interior nul byte");
         let path = b"/System/Library/Frameworks/OpenGLES.framework/OpenGLES\0";
         let addr = unsafe {
@@ -307,7 +299,7 @@ impl GlContext for Context {
     }
 
     #[inline]
-    fn swap_buffers(&self) -> Result<(), ContextError> {
+    pub fn swap_buffers(&self) -> Result<(), ContextError> {
         unsafe {
             let res: BOOL = msg_send![self.eagl_context, presentRenderbuffer:gles::RENDERBUFFER];
             if res == YES {
@@ -321,12 +313,12 @@ impl GlContext for Context {
     }
 
     #[inline]
-    fn get_api(&self) -> Api {
+    pub fn get_api(&self) -> Api {
         Api::OpenGlEs
     }
 
     #[inline]
-    fn get_pixel_format(&self) -> PixelFormat {
+    pub fn get_pixel_format(&self) -> PixelFormat {
         let color_format = ColorFormat::for_view(self.view);
         PixelFormat {
             hardware_accelerated: true,
@@ -342,8 +334,14 @@ impl GlContext for Context {
     }
 
     #[inline]
-    fn resize(&self, _width: u32, _height: u32) {
+    pub fn resize(&self, _width: u32, _height: u32) {
         // N/A
+    }
+}
+
+impl Drop for Context {
+    fn drop(&mut self) {
+        let _: () = unsafe { msg_send![self.eagl_context, release] };
     }
 }
 

